### PR TITLE
Remove in-memory sippy impl and the legacy API

### PR DESCRIPTION
**Pull request by dgoodwin**
Remove in-memory sippy impl and the legacy API.
    
    PostgreSQL sippy has been in the wild for some time now and running
    well, we are ready to jettison all the old code.
    
    Unfortunately required removing the server smoke test as we can no
    longer run without a db, and we do not yet have a way to fire up a
    temporary db for a server. This gap will need to be handled separately
    but this did not catch much anyhow.
    
    Includes removal of a duplicated testidentification directory that got
    missed somehow.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -226,10 +226,6 @@ func (o *Options) Validate() error {
 		return fmt.Errorf("must specify --database-dsn with --load-database")
 	}
 
-	if o.DBOnlyMode && o.DSN == "" {
-		return fmt.Errorf("must specify --database-dsn with --db-only-mode")
-	}
```
Comment (stbenjam) [inline main.go:231]: What do you think about having an error if DBOnlyMode is false?

File: main.go [hunk 2]
```diff
@@ -226,10 +226,6 @@ func (o *Options) Validate() error {
 		return fmt.Errorf("must specify --database-dsn with --load-database")
```
Comment (stbenjam) [inline main.go:226]: Won't database-dsn always be required now? Maybe we should error if not specified?

File: main.go (hunk 1 above)
Comment (dgoodwin) [reply main.go:231]: Yeah that should work, and still possible to edit the manifests as we'll be defaulting to true per the above recommendation.

File: main.go (hunk 2 above)
Comment (dgoodwin) [reply main.go:226]: I don't think we need it for fetch data, but I can expand this to server and loaddatabase. Seem right?

**Comment by dgoodwin** [comment]
Should I nuke the CLI report mode too? Does anyone use that? (running binary without --server) It looks busted anyhow, I'm going to add another commit to remove that as well.

**Comment by dgoodwin** [comment]
I've made the requested edits, also found a bunch more code that could be removed by following the CLI report path and removing that as well. I think it was already broken anyhow.

**Comment by stbenjam** [comment]
Checked again and I think it's all working correctly in db-only mode.

**Comment by stbenjam** [comment]
@dgoodwin Tests are failing. I think because you need to rebase. Your payload pr used strconv in server.go, but this PR removed it. Once this is rebased on master it breaks

**Comment by dgoodwin** [comment]
Thanks, I think I've addressed and squashed into appropriate commit.

**Comment by stbenjam** [comment]
Oof, something else merged again that broke this.  Let's fix this and not merge anything else until this merges :)

**Comment by dgoodwin** [comment]
Ok updated again, lets hope this is it :)

**Pull request merged**